spec/lib/finish_unfinished_persons_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FinishUnfinishedPersons, type: :module do
+  describe '.compute_semi_id' do
+    let(:competition_year) { 2023 }
+    let(:available_per_semi) { {} }
+
+    context 'with a simple name' do
+      let(:person) { create(:person, name: 'John Smith') }
+
+      it 'generates a semi_id with the correct format' do
+        allow(FinishUnfinishedPersons).to receive(:extract_roman_name).with('John Smith').and_return('John Smith')
+        allow(FinishUnfinishedPersons).to receive(:remove_accents).with('John Smith').and_return('John Smith')
```

Comment:
Why do you have to mock this behavior? Doesn't that defeat the purpose of testing that the inner workings of these methods do what you would expect them to?